embassy-nxp/src/usart/lpc55.rs

```diff
@@ -0,0 +1,854 @@
+use core::marker::PhantomData;
+use embassy_hal_internal::{Peri, PeripheralType};
+use embedded_io::{self, ErrorKind};
+pub use sealed::SealedInstance;
+use crate::gpio::AnyPin;
+
+
+/// Serial error
+#[derive(Debug, Eq, PartialEq, Copy, Clone)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[non_exhaustive]
+pub enum Error {
+    /// Triggered when the FIFO (or shift-register) is overflowed.
+    Overrun,
+    /// Triggered when a break is received
+    Break,
+    /// Triggered when there is a parity mismatch between what's received and
+    /// our settings.
+    Parity,
+    /// Triggered when the received character didn't have a valid stop bit.
+    Framing,
+    /// Triggered when the receiver detects noise?
+    Noise,
+}
+
+
+impl embedded_io::Error for Error {
+    fn kind(&self) -> ErrorKind {
+        match self {
+            Error::Overrun => ErrorKind::Other,
+            Error::Parity => ErrorKind::InvalidData,
+            Error::Framing => ErrorKind::InvalidData,
+            Error::Break => ErrorKind::Interrupted,
+            Error::Noise => ErrorKind::Other,
+        }
+    }
+}
+/// Word length.
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub enum DataBits {
+    /// 7 bits.
+    #[doc = "7 bit Data length."]
+    DataBits7,
+    /// 8 bits.
+    #[doc = "8 bit Data length."]
+    DataBits8,
+    /// 9 bits.
+    #[doc = "9 bit data length. The 9th bit is commonly used for addressing in multidrop mode."]
+    DataBits9,
+}
```

Comment:
I think we could remove the `#[doc = ...]` attribute and let the longer messages be the actual doc comment. It should be the same thing more or less. Plus the longer, more explanation-filled docs aren't that big.

Same comment for the other enums as well

```suggestion
pub enum DataBits {
    /// 7 bits data length
    DataBits7,
    /// 8 bits data length
    DataBits8,
    /// 9 bit data length. The 9th bit is commonly used for addressing in multidrop mode
    DataBits9,
}
```